packages/container/lib/container.js

```diff
@@ -618,6 +647,18 @@ class FactoryManager {
     this.normalizedName = normalizedName;
     this.madeToString = undefined;
     this.injections = undefined;
+
+    if (isObject(factory)) {
+      setFactoryManager(factory, this);
```

Comment:
factory here doesn't not have a 1 to 1 with FactoryManager, FactoryManager has only one factory but a factory can have multiple managers.